src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs

```diff
@@ -156,6 +156,53 @@ public void ResolvesServiceMixedServiceAndOptionalStructConstructorArgumentsReli
             }
         }
 
+        public enum TheEnum
+        {
+            HelloWorld = -1,
+            NiceWorld = 0,
+            GoodByeWorld = 1,
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ResolvesConstantValueTypeServicesCorrectly(bool useScoped)
```

Comment:
Add some indirect cases as well (unless we have those already).

Reply:
Looks like we have one that is currently disabled. Opened https://github.com/dotnet/runtime/issues/42160 to fix it.